File: CS/WinSample.Module/Item.cs
using DevExpress.Xpo;

using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Base.General;

namespace WinSample.Module {
    [DefaultClassOptions]
    public class Item : BaseObject, ICategorizedItem {
        public Item(Session session) : base(session) { }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
        private Category _Category;
        public Category Category {
            get { return _Category;}
            set { SetPropertyValue("Category", ref _Category, value); }
        }
        #region ICategorizedItem Members
        ITreeNode ICategorizedItem.Category {
            get {
                return Category;
            }
            set {
                Category = (Category)value;
            }
        }
        #endregion
    }
}

File: VB/WinSample.Module/Item.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Base.General

Namespace WinSample.Module
	<DefaultClassOptions> _
	Public Class Item
		Inherits BaseObject
		Implements ICategorizedItem
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property
		Private _Category As Category
		Public Property Category() As Category
			Get
				Return _Category
			End Get
			Set(ByVal value As Category)
				SetPropertyValue("Category", _Category, value)
			End Set
		End Property
		#Region "ICategorizedItem Members"
		Private Property ICategorizedItem_Category() As ITreeNode Implements ICategorizedItem.Category
			Get
				Return Category
			End Get
			Set(ByVal value As ITreeNode)
				Category = CType(value, Category)
			End Set
		End Property
		#End Region
	End Class
End Namespace
